feat: Initial commit of CIS module (#349)
terraform-ibm-modules/terraform-ibm-cis: README.md

# IBM Cloud Internet Services (CIS) Module

> This module provisions an IBM Cloud Internet Services (CIS) instance. The module includes the sub-modules to add the following features to the CIS instance.

Reviewer: missed `submodules` (not hyphenated).

Author: done. Thanks Allen.

# IBM Cloud Internet Services (CIS) Module

> This module provisions an IBM Cloud Internet Services (CIS) instance and allows to add the following into the CIS instance.

Reviewer 2: It is not fully accurate as the module itself does not allow adding capabilities, rather that is submodules in this repository. I think this should be named indicating that this repo contains a collection of modules.

Author: Indicated about submodules.

> * Domain

Reviewer 2: I would move up the domain config as it is a fundamental resource at the root module level when creating a cis instance. Alternative is to move the instance creation as a submodule and leave the root as empty (no module)

Author: 1. `Root module as empty` - had discussion on this before. Generally in goldeneye modules we do not follow root modules as empty, so took this approach.
2. `domain can be moved into root module.` yes, it can be done. But when I created only instance, it worked. Infact if I do through UI, then also first step is instance creation and then domain and other things follow. So it is like - first step is instance creation and then configure the instance (via submodules) starting from domain. So I think keeping it as sub-module is fine and matches with the UI behavior.

Let me know your thoughts ...accordingly it can be changed.

Reviewer 2: Yes, the main concern here is that the root module is just a simple wrapper around the provider resource, which is generally an anti-pattern. I think option 2 conceptually makes sense as I do not think the CIS instance is usable without domain config. Thinking further, perhaps the root could just call the domain submodule as opposed to moving up the logic. This would give a more compelling starting point at the root rather than just a thin wrapper.

Author: ok, understood the point. Made the changes as suggested.

      content = "1.2.3.4"
      ttl     = 900
    }
  ]
}

module "cis_glb" {
  source             = "terraform-ibm-modules/cis/ibm//glb"
  cis_instance_id    = module.cis_instance.cis_instance_id
  domain_id          = module.cis_domain.cis_domain.domain_id
  glb_name           = "cis_glb"
  fallback_pool_name = "cis_fpn"
  glb_enabled        = true
  origin_pools       = [
    {
      name = "glb1"
      origins = [{
        name    = "o-1"
        address = "1.1.1.0"
        enabled = true
        }]
      enabled           = true
      description       = "Test GLB"
      check_regions     = ["WEU"]
      health_check_name = "hc1"
    }
  ]
  health_checks       = [
    {
      expected_body  = "alive"
      expected_codes = "200"
      method         = "GET"
      timeout        = 7
      path           = "/health"
      interval       = 60
      retries        = 3
      name           = "hc1"
    }
  ]
}
```

## Required IAM access policies

You need the following permissions to run this module.

- Account Management
  - **Resource Group** services
    - `Editor` platform access
- IAM Services
  - **Cloud Internet** service
    - `Editor` platform access

## Examples

> * [Complete example that creates CIS instance and add domain, dns records, global load balancer](examples/complete)

Author: Updated.

# IBM Cloud Internet Services (CIS) Module

This module provisions an IBM Cloud Internet Services (CIS) instance.

> It includes the sub-modules that allows to add the following into the CIS instance.

Reviewer: ```suggestion
The module includes submodules to add the following features to the CIS instance.
```

I think you can move this 2nd sentence to line 3 to be part of the same paragraph.

Author: done
